sidecar/src/agent_remote_config.rs

```diff
@@ -0,0 +1,212 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use datadog_ipc::platform::{FileBackedHandle, MappedMem, NamedShmHandle, ShmHandle};
+use ddcommon::Endpoint;
+use std::hash::{Hash, Hasher};
+use std::io;
+use std::io::Write;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Mutex;
+use zwohash::ZwoHasher;
+
+pub struct AgentRemoteConfigWriter<T>
+where
+    T: FileBackedHandle<T> + From<MappedMem<T>>,
+{
+    handle: Mutex<Option<MappedMem<T>>>,
+}
+
+pub struct AgentRemoteConfigReader<T>
+where
+    T: FileBackedHandle<T> + From<MappedMem<T>>,
+{
+    handle: Option<MappedMem<T>>,
+    endpoint: Option<Endpoint>,
+    current_config: Option<Vec<u8>>,
+}
+
+#[repr(C)]
+#[derive(Debug)]
+struct RawMetaData {
+    generation: u64,
```

Comment:
Shouldn't this also be an atomic? It's read outside of section gated by `writing`

Reply:
Not sure if necessary, but can't do harm to make it atomic.